api/v1alpha1/imagerepository_types.go

```diff
@@ -49,6 +49,20 @@ type ImageRepositorySpec struct {
 	// equivalent.
 	SecretRef *corev1.LocalObjectReference `json:"secretRef,omitempty"`
 
+	// CertSecretRef can be given the name of a secret containing
+	// either or both of
+	//
+	//  - a PEM-encoded client certificate (`certFile`) and private
+	//  key (`keyFile`);
+	//  - a PEM-encoded CA certificate (`cacertFile`)
+	//
+	//  and whichever are supplied, will be used for connecting to the
+	//  registry. The client cert and key are useful if you are
+	//  authenticating with a certificate; the CA cert is useful if
+	//  you are using a self-signed server certificate.
+	// +optional
+	CertSecretRef *corev1.LocalObjectReference `json:"certSecretRef,omitempty"`
```

Comment:
Please use `LocalObjectReference ` from `fluxcd/pkg/apis/meta` package.

Reply:
You need to rebase first to get https://github.com/fluxcd/image-reflector-controller/commit/539409d3db591d6e503952425362968e61518999

Reply:
Doing so seems to break code generation :-(

```
$ make generate manifests api-docs 
go: creating new go.mod: module tmp
go: found sigs.k8s.io/controller-tools/cmd/controller-gen in sigs.k8s.io/controller-tools v0.3.0
cd api; /Users/mikeb/go/bin/controller-gen object:headerFile="../hack/boilerplate.go.txt" paths="./..."
230:20: missing ',' in argument list
235:20: missing ',' in argument list
Error: not all generators ran successfully
run `controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./... -w` to see all available markers, or `controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./... -h` for usage
make: *** [generate] Error 1
```

Reply:
Are they used anywhere else?

Reply:
nvm -- updating controller-gen gave me better errors (better than line numbers without a file!), making it obvious that I needed to `(cd api; go get github.com/fluxcd/apis/meta)`.